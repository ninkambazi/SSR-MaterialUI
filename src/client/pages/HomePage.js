import React from 'react';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

export class Home extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			open: false
		};
		this.handleToggle = this.handleToggle.bind(this);
		this.handleClose = this.handleClose.bind(this);
	}

	handleToggle() {
		this.setState({
			open: !this.state.open
		});	
	} 

	handleClose() {
		this.setState({
			open: false
		});
	}	

	render() {
		return (
			<div>
				<div>
					<AppBar
					    title="RoadPiper Admin Panel"
					    onLeftIconButtonClick={this.handleToggle}
					 />
					 <Drawer
					 	docked={false}
					 	open={this.state.open}
					 	onRequestChange={(open) => this.setState({open})}
					 >
					 	<MenuItem>Menu Item 1</MenuItem>
					 	<MenuItem>Menu Item 2</MenuItem>
					 </Drawer>
				</div> 
			</div>
		);
	}
};

export default {
	component: Home
};