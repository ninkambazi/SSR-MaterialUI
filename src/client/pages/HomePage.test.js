import React from 'react';
import { Home } from './HomePage';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

describe('Home', () => {
	let wrapper

	it('1. Renders the HomePage without crashing', () => {
		wrapper = shallow(<Home />);
		expect(wrapper).toHaveLength(1);
	})

	it('2. One <AppBar/> Component is present', () => {
		wrapper = shallow(<Home />);
		expect(wrapper.find(AppBar)).toHaveLength(1);
	})

	it('3. One <Drawer/> Component is present', () => {
		wrapper = shallow(<Home />);
		expect(wrapper.find(Drawer)).toHaveLength(1); 
	})

	it('4. <AppBar/> title should be RoadPiper Admin Panel', () => {
		wrapper = shallow(<Home />);
		expect(wrapper.find(AppBar).prop('title')).toBe('RoadPiper Admin Panel');
	})

	it('5. <Drawer/> Component has two Menu Items present', () => {
		wrapper = shallow(<Home />);
		expect(wrapper.find(Drawer).find(MenuItem)).toHaveLength(2);
	})

	it('6. Should match the snapshot', () => {
		wrapper = shallow(<Home />);
		expect(wrapper).toMatchSnapshot();
	})

	it('7. <AppBar/> should have button to open <Drawer/>', () => {
		wrapper = shallow(<Home />);
		expect(wrapper.find(AppBar).prop('onLeftIconButtonClick')).toBeTruthy;
	})

	it('8. <Drawer/> opens when Icon Button is clicked', () => {
		wrapper = shallow(<Home />);
		wrapper.find(AppBar).prop('onLeftIconButtonClick')('click');
		expect(wrapper.find(Drawer)).toHaveLength(1);
	})
})