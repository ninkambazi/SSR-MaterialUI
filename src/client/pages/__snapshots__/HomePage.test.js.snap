// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 6. Should match the snapshot 1`] = `
<div>
  <div>
    <AppBar
      onLeftIconButtonClick={[Function]}
      showMenuIconButton={true}
      title="RoadPiper Admin Panel"
      zDepth={1}
    />
    <Drawer
      disableSwipeToOpen={false}
      docked={false}
      onRequestChange={[Function]}
      open={false}
      openSecondary={false}
      swipeAreaWidth={30}
      width={null}
      zDepth={2}
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Menu Item 1
      </MenuItem>
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Menu Item 2
      </MenuItem>
    </Drawer>
  </div>
</div>
`;
